### PR TITLE
PLU-673 (feat): helpers to parse clarifications

### PR DESCRIPTION
**Pull request by kevinkim-ogp**
### **TL;DR**

Adds a parser for extracting structured clarification questions from LLM responses, along with a new `data-clarification` message part type in the chat schema.

### **What changed?**

- Introduced `parseClarificationBlock`, which extracts `ClarificationQuestion` objects (each with a `question` string and at least two `options`) from a `CLARIFICATION_DATA` HTML comment block embedded in LLM response text. Returns `null` when no valid block is found.
- Added a `data-clarification` discriminated union variant to the chat message part schema, carrying an array of questions with their options.
- Added comprehensive unit tests covering empty blocks, missing questions, single/multiple questions, stray lines, whitespace handling, and insufficient options.

### **How to test?**

- [ ] Sanity check that the parsers make sense
- [ ] Verify all test cases pass, including edge cases such as blocks with fewer than 2 options per question (should return `null`) and blocks with surrounding whitespace (should parse correctly).

### **Why make this change?**

When the LLM needs more information before generating a workflow, it should be able to surface structured clarification questions to the user. This parser provides a reliable way to extract those questions from the raw LLM response text, and the new schema type ensures the clarification data can be properly validated and transmitted through the chat message pipeline.

This is in preparation for the next PR, which introduces a new frontend component to collect the responses to the clarifications.

### CODE REVIEW

**Comment by kevinkim-ogp** [comment]
* **#1648** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1648?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1647** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1647?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1647?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1644** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1644?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1622** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1622?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1621** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1621?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1608** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1608?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1607** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1607?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>lfg</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/opengovsg?ref=merge-queue-instructions-in-stack-comment&prId=1647).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/backend/src/routes/api/chat/parse-clarification-block.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      if (current) {
+        questions.push(current)
+      }
+      const question = line.slice(2).trim()
+      if (!question) {
+        current = null
+        continue
+      }
+      current = { question, options: [] }
+    } else if (line.startsWith('-') && current) {
```
Comment (ogp-weeloong) [inline packages/backend/src/routes/api/chat/parse-clarification-block.ts]: this could be dangerous if our robot wraps questions to a new line, although i think it doesnt' do that now.  


example:

```
Q: What trigger? For example, the postman
-transactional-email action is usually
used with FormSG Trigger
- FormSG
- Webhook
```

File: packages/backend/src/routes/api/chat/schema.ts [hunk 2]
```diff
@@ -32,6 +32,17 @@ const messagePartSchema = z.discriminatedUnion('type', [
...
   }),
+  z.object({
+    type: z.literal('data-clarification'),
+    data: z.object({
+      questions: z.array(
+        z.object({
+          question: z.string(),
+          options: z.array(z.string()).min(2),
+        }),
+      ),
```
[the remark this answers is not in the document]
Comment (ogp-weeloong) [reply packages/backend/src/routes/api/chat/schema.ts]: this quite smart, I think it actually caught a thing. @kevinkim-ogp

**Review by ogp-weeloong — APPROVED** [review]
looks good!
